pkgs/stdenv/generic/make-derivation.nix

```diff
@@ -12,7 +12,9 @@ rec {
   # * https://nixos.org/nix/manual/#ssec-derivation
   #   Explanation about derivations in general
   mkDerivation =
-    { name ? ""
+    { name ? if builtins.hasAttr "pname" attrs && builtins.hasAttr "version" attrs
```

Comment:
Can you update "pos" above as well? It is used to find line numbers in traces. I think we need to also look for "version" before name for this to work correctly.

Reply:
Would that be something like `builtins.unsafeGetAttrPos "pname" attrs`?

Reply:
Yeah but I actually think version would be better because it's more frequently used than pname.

Reply:
Okay, I'll add it.

Reply:
Does 0c1d5d1 look good to you?

Reply:
Oh, yeah, that would make sense. Adding that.

Reply:
@globin e0d2348

Reply:
This should be `attrs ? pname && attrs ? version`.